Sources/MongoSwift/Operations/ListDatabasesOperation.swift

```diff
@@ -73,19 +73,21 @@ internal struct ListDatabasesOperation: Operation {
         var reply = Document()
         var error = bson_error_t()
 
-        let success = cmd.withBSONPointer { cmdPtr in
-            readPref.withMongocReadPreference { rpPtr in
-                withOptionalBSONPointer(to: opts) { optsPtr in
-                    reply.withMutableBSONPointer { replyPtr in
-                        mongoc_client_read_command_with_opts(
-                            connection.clientHandle,
-                            "admin",
-                            cmdPtr,
-                            rpPtr,
-                            optsPtr,
-                            replyPtr,
-                            &error
-                        )
+        let success = connection.withMongocConnection { connPtr in
+            cmd.withBSONPointer { cmdPtr in
+                readPref.withMongocReadPreference { rpPtr in
+                    withOptionalBSONPointer(to: opts) { optsPtr in
+                        reply.withMutableBSONPointer { replyPtr in
+                            mongoc_client_read_command_with_opts(
+                                connPtr,
+                                "admin",
+                                cmdPtr,
+                                rpPtr,
+                                optsPtr,
+                                replyPtr,
+                                &error
+                            )
+                        }
```

Comment:
\*_dreaming of pure swift_\* one day we won't have to nest this deep 😄

Reply:
ooh, I like how you can comment on ranges of lines now.